### ISSUE TITLE
Add get data as array option

### ISSUE THREAD

**Issue by tamtakoe** [firebase/angularfire#200]
Getting collection ad object is very unconfortable. Use orderByPriority too. Make option that collection get always like array.

Also make API-compatible resourse with angular resource

**Comment by mastef** [comment]
I've posted a [gist](https://gist.github.com/mastef/0ae8ba7595cafad2c85f/revisions?diff=unified) at the [linked issue](https://github.com/angular-ui/ui-sortable/issues/421), that works with moving items between 2 $firebaseArrays within ui-sortable.

I'm not sure it's the best practice, and from intense googling @katowulf is _the_ expert on firebaseArray issues. It would be supercool if kato could review if that's a good way of handling the firebaseArrays then I think we'd be a step closer for native firebase support in ui.sortables :-)

Basically removed any slice options, except one since it's used only for re-ordering of a single list, and later on priorities are updated anyway based on indexOf position.

**Comment by katowulf** [comment]
Thanks! I've put this on my todo list. There's a lot to read through here to get up to speed on all the nuances and possible solutions. Since the diff of the gist is mostly green, could you highlight the key points? That would help shave a bit of angle off of the learning curve.

**Comment by katowulf** [comment]
Could you also open a separate issue rather than multi-purposing a closed, old task?

**Comment by mastef** [comment]
Hey @katowulf thanks for the reply and happy new year! I've posted it into #687
